.github/workflows/release.yml

```diff
@@ -0,0 +1,53 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Build package
+    runs-on: ubuntu-latest
+    env:
+      PYTHON_VERSION: ${{ secrets.PYTHON_VERSION }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ vars.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ vars.PYTHON_VERSION }}
+
+      - name: Install dependencies
+        run: python -m pip install -U build
+
+      - name: Build package
+        run: python -m build
+
+      - name: Upload dist folder
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist_directory
+          path: dist
+  
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    needs: build
+    env:
+      TOKEN: ${{ secrets.pypi_token }}
+    steps:
+      - name: Download dist folder
+        uses: actions/download-artifact@v4
+        id: download
+        with:
+          name: dist_directory
+          path: dist
+      - name: Publish package to PyPI 
+        if: steps.download.conclusion == 'success' && env.TOKEN != '' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: ${{ secrets.pypi_user }}
+          repository-url: https://test.pypi.org/legacy/
```

Comment:
This is not our official package but points to a test instance of pypi.

Reply:
Thank you. When uploading your package, you'll need to specify a repository URL. There are two main options: TestPyPi and PyPi. It's highly recommended to upload your package to TestPyPi (https://test.pypi.org/) first, before uploading to another. Note that the credentials for these two repositories are different. Do you have a custom repository/registry that you're using? I followed the guide on packaging.python.org for reference.